lib/rsvp/enumerator.js

```diff
@@ -39,24 +40,23 @@ function Enumerator(Constructor, input, abortOnReject, label) {
   this.promise = new Constructor(noop, label);
   this._abortOnReject = abortOnReject;
 
-  if (this._validateInput(input)) {
-    this._input     = input;
-    this.length     = input.length;
-    this._remaining = input.length;
+  if (!this._validateInput(input)) {
+    return reject(this.promise, this._validationError());
+  }
+
+  this._input     = input;
```

Comment:
I believe this where set in the above `if` statement, so that all instance of enumerator (rejected or not) would have the same shape.